### PR TITLE
Add user customer domain

### PR DESCRIPTION
**Pull request by umar006**

### CODE REVIEW

File: internal/domain/user-customer.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/google/uuid"
+)
+
+type UserCustomer struct {
+	ID          string    `json:"id" db:"id"`
+	CreatedAt   time.Time `json:"createdAt" db:"created_at"`
+	Name        string    `json:"name" db:"name"`
+	PhoneNumber string    `json:"phoneNumber" db:"phone_number"`
+	Password    string    `json:"password" db:"password"`
+}
```
Comment (fathisiddiqi) [inline internal/domain/user-customer.go]: domain buat usecase harusnya gak ada tag json

File: internal/domain/user-customer.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	createdAt, _ := time.Parse(time.RFC3339, rawCreatedAt)
+
+	return UserCustomer{
+		ID:          id.String(),
+		CreatedAt:   createdAt,
+		Name:        body.Name,
+		PhoneNumber: body.PhoneNumber,
+		Password:    body.Password,
+	}
+}
```
Comment (fathisiddiqi) [inline internal/domain/user-customer.go:41]: https://gin-gonic.com/docs/examples/binding-and-validation/

File: internal/domain/user-customer.go (hunk 1 above)
Comment (umar006) [reply internal/domain/user-customer.go]: done

File: internal/domain/user-customer.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	PhoneNumber string `json:"phone_number"`
+	Password    string `json:"password"`
+}
+
+type RegisterUserCustomerResponse struct {
+	ID          string `json:"id"`
+	Name        string `json:"name"`
+	PhoneNumber string `json:"phoneNumber"`
+}
+
```
Comment (fathisiddiqi) [inline internal/domain/user-customer.go:28]: login user request sama responsenya mar sekalian

File: internal/domain/user-customer.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+type GetUserCustomerResponse struct {
+	ID          string `json:"id"`
+	Name        string `json:"name"`
+	PhoneNumber string `json:"phoneNumber"`
+}
```
Comment (fathisiddiqi) [inline internal/domain/user-customer.go]: CustomerResponse aja mar ini

Comment (umar006) [reply internal/domain/user-customer.go]: `UserCustomerResponse`, biar konsisten dgn yg lain

**Review by fathisiddiqi — APPROVED** [review]

**Pull request merged**